### ISSUE TITLE
The impact of topk pseudo-labels selection for unknown training?

### ISSUE THREAD

**Issue by harrylin-hyl** [orrzohar/PROB#10]
I noticed that you do not use the top 5 pseudo-labels (which are not overlapping with the ground truth) to train the unknown detection.
Could you please tell me why? Because most of the OWOD methods such as OW-DETR adopt this strategy.

Best regards,
Yulin

**Comment by orrzohar** [comment]
Hi Yulin,
This is because PROB does not use pseudo-labels. Rather than having the classification head directly predict unknown objects using pseudo-labeling `supervision', PROB splits the inference problem intro _objectness_ prediction and _classification_. I.e., rather than having a classification head with effectively C+2 logits, where there are C known classes, the C+1 logit is unknown objects and the C+2 logit is background, PROB uses C+1 logits where the C+1 logit is unknown object _and_ background.

Now, to `filter' out and/or suppress the unknown object predictions, we use the separate objectness head, which knows only to separate objects (both known and unknown) and backgrounds. Specifically, for PROB, p_label = p_object * p_cls.

The motivation for this is that now we split one unsupervised problem (i.e., for an unlabeled proposal, one needs to determine if it is an unknown object - C+1 -  or background -C+2) into a fully semi-supervised one (objectness + regular classification). 

Hope this clears this up,
Orr

**Comment by harrylin-hyl** [comment]
Thanks for your quick reply! I got it!
